app/service/rest.py

```diff
@@ -210,6 +218,82 @@ def get_service_notification_statistics(service_id):
     )
 
 
+@service_blueprint.route("/<uuid:service_id>/statistics/<string:start>/<int:days>")
+def get_service_notification_statistics_by_day(service_id, start, days):
+    return jsonify(
+        data=get_service_statistics_for_specific_days(service_id, start, int(days))
+    )
+
+
+def get_service_statistics_for_specific_days(service_id, start, days=1):
+    # start and end dates needs to be reversed because
+    # the end date is today and the start is x days in the past
+    # a day needs to be substracted to allow for today
+    end_date = datetime.strptime(start, "%Y-%m-%d")
+    start_date = end_date - timedelta(days=days - 1)
+
+    results = dao_fetch_stats_for_service_from_days(service_id, start_date, end_date)
+
+    grouped_results = defaultdict(list)
+    for row in results:
+        notification_type, status, day, count = row
+        grouped_results[day.date()].append(row)
+
+    for date in generate_date_range(start_date, days=days):
+        if date not in grouped_results:
+            grouped_results[date] = []
+
+    stats = {}
+    for day, rows in grouped_results.items():
+        stats[day.strftime("%Y-%m-%d")] = statistics.format_statistics(rows)
+
```

Comment:
A little more efficient and cleaned up would be this.
```
def get_status(start_date, days, results):
    grouped_results = {
        date: []
        for date in generate_date_range(start_date, days=days
    } | {
        day.date(): [notification_type, status, day, count]
        for notification_type, status, day, count in results
    }

    stats = {
        day.strftime("%Y-%m-%d"): statistics.format_statistics(rows)
        for day, rows in grouped_results.items()
    }
    
    return stats
```

Then simply use this in the place of the for loops, to get the statistics, and have it uniformly done.

Comprehensions are thousands of times faster & more efficient as they are close to directly C loops without the baggage of for/if Python operators. However, it could be argued that your method is a smidge more readable for people who are unfamiliar with Python. Still, this would be more idiomatic/pythonic.